### PR TITLE
doc and lint revisions

### PR DESCRIPTION
**Pull request by jessp01**
- Some typo corrections
- Doc line above exported consts and functions
- Implemented some golint suggestions

I also added `.pre-commit-config.yaml` to the repo. To use it:

- `pip install pre-commit`
- Install the following Go packages:
  * go install github.com/go-critic/go-critic/cmd/gocritic@latest
  * go install https://pkg.go.dev/golang.org/x/tools/cmd/goimports@latest
  * go install golang.org/x/lint/golint@latest
- In your clone, run: `pre-commit install`

This will create .git/hooks/pre-commit and following that, these tests will run every time you invoke `git commit`:
```sh
go fmt...................................................................Passed
go imports...............................................................Passed
go vet...................................................................Passed
go lint..................................................................Passed
go-critic................................................................Passed
```

For now, I commented the `go-imports`, `go-vet`, `go-lint` and `go-critic` as they still have findings that should be looked it but I recommend enabling these once that's done.

To run all checks, uncomment them in `.pre-commit-config.yaml`, invoke `pre-commit install` and run `pre-commit run --all-files`

Attached is the full output of such a run.
[melange-lint.log](https://github.com/chainguard-dev/melange/files/12307877/melange-lint.log)

### CODE REVIEW

**Review by kaniini — APPROVED** [review]

**Comment by kaniini** [comment]
> I also added .pre-commit-config.yaml to the repo.

It does not actually seem to be present in this PR.

File: pkg/cli/update_cache.go [hunk 1]
```diff
@@ -15,15 +15,14 @@
 package cli
 
 import (
-	_ "context"
-	_ "fmt"
```
Comment (kaniini) [inline pkg/cli/update_cache.go:18]: To appease `go imports` check:

```suggestion
```

**Comment by kaniini** [comment]
Please run `go fmt ./...` on the code.  Also, please flatten the commits, grouping them by concern.

**Comment by jessp01** [comment]
I'm not seeing any `gofmt`  warnings in any of my changed files (or any files in this repo for that matter). Which file are you referring to?

**Comment by kaniini** [comment]
https://github.com/chainguard-dev/melange/actions/runs/5815734548/job/15767805770?pr=598

**Review by kaniini — APPROVED** [review]

**Comment by jessp01** [comment]
Yeap, I saw that failure and fixed it here https://github.com/chainguard-dev/melange/pull/598/commits/1872fa5116d5a4cb879919f2dccf498857306939

**Comment by kaniini** [comment]
Ok, looks fine.  I have decided to just set the auto-merge to do a squash, so don't worry about the commits.

**Pull request merged**